src/audio/codec_adapter/codec_adapter.c

```diff
@@ -326,6 +332,72 @@ int codec_adapter_copy(struct comp_dev *dev)
 	comp_dbg(dev, "codec_adapter_copy() start: codec_buff_size: %d, local_buff free: %d source avail %d",
 		 codec_buff_size, local_buff->stream.free, source->stream.avail);
 
+	/* compute number of input buffers */
+	list_for_item(blist, &dev->bsource_list)
```

Comment:
is there a list_len macro we can import from the kernel?

Reply:
I dont think there's one. I moved this part to the prepare() callback now so that we dont do this for every copy